src/db/models/user.model.js

```diff
@@ -172,5 +264,40 @@ module.exports = (sequelize, DataTypes) => {
             });
     };
 
+    User.prototype.updateVerifyAccountToken = function updateVerifyAccountToken(email, expTime) {
+        return randomBytes(20)
+            .then((buf) => {
+                const token = buf.toString('hex');
+                return token;
+            }).then((token) => {
+                this.contactMethodVerificationToken = token;
+                this.contactMethodToVerify = email;
+                const m = moment.utc();
+                m.add(expTime, 'seconds');
+                this.contactMethodVerificationTokenExpires = m.format();
+                return this.save().then(() => token);
+            });
+    };
+
+    User.prototype.addVerifiedContact = function addVerifiedContact(contactMethodToVerify) {
+        if (this.verifiedContactMethods.includes(contactMethodToVerify)) {
+            // Handle instances where the user contact already exsits in the list of verified
+            // contact methods
+            this.contactMethodVerificationToken = null;
+            this.contactMethodVerificationTokenExpires = null;
+            this.contactMethodToVerify = null;
+            return this.save();
+        }
+        const authorizedUsers = this.verifiedContactMethods;
+        const authorizedUsersLength = authorizedUsers.push(contactMethodToVerify.toString());
```

Comment:
It seems it would be more straightforward if the `.push` was at the end of line `const authorizedUsers = this.verifiedContactMethods;`

Reply:
I was getting an error related to the way pushing to authorized users would return a length value. So this was a way around that.